controllers/vrg_recipe.go

```diff
@@ -0,0 +1,259 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/go-logr/logr"
+	ramen "github.com/ramendr/ramen/api/v1alpha1"
+	"github.com/ramendr/ramen/controllers/kubeobjects"
+	"github.com/ramendr/ramen/controllers/util"
+	recipe "github.com/ramendr/recipe/api/v1alpha1"
+	"golang.org/x/exp/slices"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+type RecipeElements struct {
+	PvcSelector     PvcSelector
+	CaptureWorkflow []kubeobjects.CaptureSpec
+	RecoverWorkflow []kubeobjects.RecoverSpec
+}
+
+func captureWorkflowDefault(vrg ramen.VolumeReplicationGroup) []kubeobjects.CaptureSpec {
+	return []kubeobjects.CaptureSpec{
+		{
+			Spec: kubeobjects.Spec{
+				KubeResourcesSpec: kubeobjects.KubeResourcesSpec{
+					IncludedNamespaces: []string{vrg.Namespace},
+				},
+			},
+		},
+	}
+}
+
+func recoverWorkflowDefault() []kubeobjects.RecoverSpec { return []kubeobjects.RecoverSpec{{}} }
+
+func GetPVCSelector(ctx context.Context, reader client.Reader, vrg ramen.VolumeReplicationGroup,
+	ramenConfig ramen.RamenConfig,
+	log logr.Logger,
+) (PvcSelector, error) {
+	var recipeElements RecipeElements
+
+	return recipeElements.PvcSelector, recipeVolumesAndOptionallyWorkflowsGet(
+		ctx, reader, vrg, ramenConfig, log, &recipeElements,
+		func(recipe.Recipe, *RecipeElements, ramen.VolumeReplicationGroup) error { return nil },
+	)
+}
+
+func RecipeElementsGet(ctx context.Context, reader client.Reader, vrg ramen.VolumeReplicationGroup,
+	ramenConfig ramen.RamenConfig, log logr.Logger, recipeElements *RecipeElements,
+) error {
+	return recipeVolumesAndOptionallyWorkflowsGet(ctx, reader, vrg, ramenConfig, log, recipeElements,
+		recipeWorkflowsGet,
+	)
+}
+
+func recipeVolumesAndOptionallyWorkflowsGet(ctx context.Context, reader client.Reader, vrg ramen.VolumeReplicationGroup,
```

Comment:
The only way to support multiple namespaces as it stands is with a recipe. This need not be a per-condition for multiple namespaces. IOW, a recipe should not be the only way multiple namespace selector should be provided as an API.

The change here would look like so:
- Add a vrg.spec.namespaces list field that states which namespaces the current VRG intends to protect
- This should also be pushed up into the DRPC for hub controlled workflows
- Use the vrg namespaces if recipe is not provided (IOW, continue the same intention that the recipe PVC selector is given a higher priority than the vrg spec selector as it currently stands)